### PR TITLE
Eslint rules tidy

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- tidy up config eslint rule
- apply it to the codebase
- rename to `eslint-plugin-guardian-frontend`

this includes the way i did it last night, just because it also involves a file move. if you'd prefer it the way we had it before, let me know and I'll update this

### CODE REVIEW

**Review by NataliaLKB — APPROVED** [review]
Yay! Its much nicer than when we looked at it last. And we even have docs 🎉 . lgtm 👍

File: tools/eslint-plugin-guardian-frontend/rules/global-config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                    const [
+                        dot1,
+                        parent,
+                        dot2,
+                        grandparent,
+                    ] = context.getSourceCode().getTokensBefore(node, 4).reverse();
+
+                    if (isDot(dot1) && isIdentifierOf('guardian', parent)
+                        && (!isDot(dot2) || isIdentifierOf('window', grandparent))
+                    ) {
```
Comment (SiAdcock) [inline tools/eslint-plugin-guardian-frontend/rules/global-config.js]: Will this catch:

```
var guardian = window.guardian;
var config = guardian.config;
```
?

**Comment by SiAdcock** [comment]
Really nice, dude. I love how many times we were importing the `common/utils/config` module and then referring to `window.guardian.config` 🎉

File: tools/eslint-plugin-guardian-frontend/rules/global-config.js (hunk 1 above)
Comment (sndrs) [inline tools/eslint-plugin-guardian-frontend/rules/global-config.js]: hmm, no, probably not..

**Pull request merged**